omas/omas_utils.py

```
@@ -7,6 +7,213 @@
+        self._error_upper = error_upper
+        assert error_lower is None or not isinstance(error_lower, str), 'error_lower must be a number'
+        self._error_lower = error_lower
+        assert error_index is None or isinstance(error_index, int), 'error_index must be a integer'
+        self._error_index = error_index
+        assert location is None or isinstance(location, str), 'location must be a string'
+        self._location = location
+
+    def __getstate__(self):
+        state = {}
+        for item in ['_nominal_value', '_error_upper', '_error_lower', '_error_index', '_location']:
+            state[item] = self.__dict__[item]
+        return state
+
+    def __setstate__(self, state):
+        for item in ['_nominal_value', '_error_upper', '_error_lower', '_error_index', '_location']:
+            self.__dict__[item] = state[item]
+        return self
+
+    @property
+    def location(self):
+        return self._location
+
+    @property
+    def nominal_value(self):
+        return self._nominal_value
+
+    @property
+    def error_upper(self):
+        if self._error_upper is not None:
+            return self._error_upper
+        else:
+            return self.nominal_value * 0.0
+
+    @property
+    def error_lower(self):
+        if self._error_lower is not None:
+            return self._error_lower
+        elif self._error_upper is not None:
+            return self._error_upper
+        else:
+            return self.nominal_value * 0.0
+
+    @property
+    def error_index(self):
+        return self._error_index
+
+    @property
+    def std_dev(self):
+        '''
+        Return standard deviation for symmetric uncertainty bounds
+        NOTE: This method will raise an error if uncertainty is not symmetrical
+        '''
+        if self._error_upper is None:
+            return self.nominal_value * 0.0
+        elif self._error_lower is None:
+            return self.error_upper
+        else:
```

What about the case where `self.error_upper==self.error_lower`?

---

Good point, but after some thought I think it is better to raise an error, the point being that if the error is symmetrical then only `error_upper` should be filled. It's better to raise these issues always, rather than having things pass in some cases but not others.

---

As long as we thought about it, and made an active decision, I am OK.